File: android-bridge/app/src/main/java/com/jaato/a11ybridge/state/SnapshotClock.kt
package com.jaato.a11ybridge.state

import java.util.concurrent.atomic.AtomicLong

/**
 * Process-wide monotonic "world version" (device design §4).
 *
 * Bumped by the settle detector on every qualifying content/state change, so that
 * `observe`, `settled`, and bundled screenshots can all stamp the same value and the
 * daemon can tell whether the world moved between two frames. Never reused, never reset.
 */
object SnapshotClock {
    private val value = AtomicLong(0)

    val current: Long get() = value.get()

    /** Advance the version; returns the new value. */
    fun bump(): Long = value.incrementAndGet()
}

File: windows-bridge/JaatoBridge/State/SnapshotClock.cs
namespace JaatoBridge.State;

/// <summary>
/// Process-wide "world version" (§8 snapshotVersion). Monotonic. Every observe/settle stamps the
/// tree (and any bundled frame) with one value so the daemon knows tree and image describe one moment.
/// </summary>
public sealed class SnapshotClock
{
    long _v;
    public long Current => Interlocked.Read(ref _v);
    public long Next() => Interlocked.Increment(ref _v);
}
